yarn-project/acir-simulator/src/client/simulator.ts

```diff
@@ -168,12 +168,23 @@ export class AcirSimulator {
     storageSlot: Fr,
     notePreimage: Fr[],
   ) {
-    let abi: FunctionAbiWithDebugMetadata;
-    try {
-      abi = await this.db.getFunctionABI(contractAddress, computeNoteHashAndNullifierSelector);
-    } catch (e) {
+    let abi: FunctionAbiWithDebugMetadata | undefined = undefined;
+
+    // Brute force
+    for (let i = 1; i < 5; i++) {
```

Comment:
I believe we got a `MAX_NOTE_FIELDS_LENGTH` constant we could use here instead of the `5`